### PR TITLE
Fix tpool in main

### PR DESCRIPTION
**Pull request by asiegel-jt**

### CODE REVIEW

**Review by jumpsiegel — APPROVED** [review]

File: src/util/tpool/fd_tpool.cxx [hunk 1]
```diff
@@ -29,12 +30,21 @@ fd_tpool_private_worker( int     argc,
   FD_VOLATILE( worker->state ) = FD_TPOOL_WORKER_STATE_BOOT;
   FD_COMPILER_MFENCE();
 
+  if( scratch_sz ) {
+    if( !scratch ) {
+      scratch = aligned_alloc( FD_SCRATCH_SMEM_ALIGN, fd_ulong_align_up( scratch_sz, FD_SCRATCH_SMEM_ALIGN ) );
```
Comment (mmcgee-jump) [inline src/util/tpool/fd_tpool.cxx:35]: Is this change necessary? Can't the caller do the aligned alloc? Having tpool take an implicit / hidden dependency on malloc is the kind of stuff we have hated about Rust.

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
Not sure about tpool allocating.

**Comment by jumpsiegel** [comment]
this will give us a pretty huge performance improvement in the ci tests... looking forward to this

File: src/util/tpool/fd_tpool.cxx (hunk 1 above)
Comment (asiegel-jt) [reply src/util/tpool/fd_tpool.cxx:35]: Worker threads need to allocate scratch space and this is the only way to do it.

**Comment by mmcgee-jump** [comment]
Can you check with Kevin if he is OK with this design?
